src/dotnet/commands/dotnet-migrate/MigrateCommand.cs

```diff
@@ -132,7 +133,15 @@ private void UpdateSolutionFile(MigrationReport migrationReport)
                 }
                 else
                 {
-                    csprojFilesToAdd.Add(Path.Combine(report.ProjectDirectory, migratedProjectName));
+                    var csprojPath = Path.Combine(relReportPath, migratedProjectName);
+                    var slnAlreadyContainsMigratedCsproj = _slnFile.Projects
+                        .Where(p => p.FilePath == csprojPath)
+                        .Any();
+
+                    if (!slnAlreadyContainsMigratedCsproj)
```

Comment:
If a folder has a csproj already and a project.json file, then migration succeeds. But if that csproj already is referenced in the sln file, then we should not try to add it to the sln file. This is the cause of the referenced issue (trying to call dotnet sln add for a project that is already referenced in the sln file).